test/modules/aws-ecs-integration.ts

```diff
@@ -111,9 +111,26 @@ describe('ECS Integration Testing', () => {
 
     // Service dependencies
     it('adds aws_service dependencies', query(`
-      call create_or_update_aws_target_group('${serviceTargetGroupName}', 'ip', ${hostPort}, 'default', 'HTTP', '/health');
-      call create_or_update_aws_load_balancer('${serviceLoadBalancerName}', 'internet-facing', 'default', 'application', 'ipv4');
-      call create_or_update_aws_listener('${serviceLoadBalancerName}', ${hostPort}, 'HTTP', 'forward', '${serviceTargetGroupName}');
+      BEGIN;
+        INSERT INTO aws_target_group
+            (target_group_name, target_type, protocol, port, vpc, health_check_path)
+        VALUES
+            ('${serviceTargetGroupName}', 'ip', 'HTTP', ${hostPort}, 'default', '/health');
+        INSERT INTO aws_load_balancer
+            (load_balancer_name, scheme, vpc, load_balancer_type, ip_address_type)
+        VALUES
+            ('${serviceLoadBalancerName}', 'internet-facing', 'default', 'application', 'ipv4');
+        INSERT INTO aws_load_balancer_security_groups
+            (aws_load_balancer_id, aws_security_group_id)
+        VALUES
+            ((SELECT id FROM aws_load_balancer WHERE load_balancer_name = '${serviceLoadBalancerName}' LIMIT 1),
+              (SELECT id FROM aws_security_group WHERE group_name = 'default' LIMIT 1));
+        INSERT INTO aws_listener
+            (aws_load_balancer_id, port, protocol, action_type, target_group_id)
+        VALUES 
+            ((SELECT id FROM aws_load_balancer WHERE load_balancer_name = '${serviceLoadBalancerName}' LIMIT 1), 
+              ${hostPort}, 'HTTP', 'forward', (SELECT id FROM aws_target_group WHERE target_group_name = '${serviceTargetGroupName}' LIMIT 1));
+      COMMIT;
```

Comment:
Does this *need* to be atomic, or are you just making sure cruft isn't left behind in the DB if it fails?

Reply:
yes, just making sure that when we execute multiple statements won't left something weird behind